ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-service-insertion.j2

```diff
@@ -7,4 +7,46 @@
 {% if router_service_insertion.enabled is arista.avd.defined(true) %}
 
 Router service-insertion is enabled.
+{%     if router_service_insertion.connections is arista.avd.defined %}
+
+## Connections
+{%         for connection in router_service_insertion.connections | arista.avd.natural_sort('name') %}
+{%             if connection.ethernet_interface.name is arista.avd.defined connection.ethernet_interface.next_hop is arista.avd.defined and connection.monitor_connectivity_host is arista.avd.defined %}
+
+### Connections Through Ethernet Interface
+
+| Name | Interface | Next Hop | Monitor Connectivity Host |
+| ---- | --------- | -------- | ------------------------- |
+{%                 break %}
```

Comment:
I think you are trying to make sure that we render only once the header.
I would prefer that you have some kind of preprocessing where 

1/ you loop over all the connections and create to sublist one for ethernet_connections and one for tunnel_connections
2/ you render ethernet connections if list ethernect_connections is not empty
3/ you render tunnel connections

Reply:
Implemented this code.